src/lib/qboard.ts

```diff
@@ -109,7 +115,7 @@ export default class QBoard {
     this.switchTool(Tool.Move);
     this.windowResize();
 
-    window.onresize = this.windowResize;
+    this.window.onresize = this.windowResize;
```

Comment:
same with the previous comment; i don't think we should make document or window an instance variable

Reply:
One issue is that this is almost like a "magic number" in a sense. It's a little worse than that; any function which needs access to the current HTML window and uses `window` instead of `this.window` is using properties that
1. are not a part of ECMAScript, only the Web API of HTML
1. can theoretically be mutated, though we don't care in this case since we assign the mutable HTML window to window anyways
1. don't generalize to more complex applications. In the specific case of qboard, maybe we don't care, but the window that controls execution is not necessarily the same as the presentation-layer window
1. can be overridden and overwritten with global scope function declarations. We want to take care to prevent this.

Additionally, I feel an instance of qboard should know where its running, especially because you provide the canvas elements when you instantiate it anyways.

Another issue is that some external method might in the future treat something that accesses the HTML `window` as a public method, but then they don't know that it has a dependency on `window`. This is like having an API which internally relies on client system time when doing so is unnecessary. It also makes the block of code not unit-testable.

Reply:
In the specific case of qboard, maybe we don't care » yes, precisely, we don’t care

i refuse to complicate the code for these reasons

Reply:
I still think point (1) is important here. You're already using typescript to make sure that you implicitly use things you're not supposed to, and the same thing's happening here. It definitely doesn't pollute the code more than converting from boolean to string does when writing to localStorage.

Reply:
i dont really think its that important

Reply:
Are there no plans to unit test this codebase?

Reply:
no

Reply:
I really don't think this is a complication to the codebase, but I'll revert it

Reply:
Resolved by f6241b1200d09ec42bea5400f67b5f6e1fb7278d